src/lib/client.spec.ts

```
@@ -15,6 +15,264 @@
+    expect(client.session.signature).toBe(options.security.signature);
+  });
+  it('should pass cname to the session', () => {
+    const options = {
+      cname: 'testCname.com',
+    };
+    const client = new Client(defaultApikey, options);
+    expect(client.session.cname).toBe(options.cname);
+  });
+  it('should be able to logout for provied cloud', (done) => {
+    const client = new Client(defaultApikey);
+    const cloudName = 'dropboxx';
+    return client.logout(cloudName).then(() => {
+      expect.assertions(2);
+      expect(mockCloudLogout).toHaveBeenCalledTimes(1);
+      expect(mockCloudLogout).toHaveBeenCalledWith(cloudName);
+      done();
+    });
+  });
+  it('should get metadata for a provided handle', (done) => {
+    const client = new Client(defaultApikey);
+    const options = {
+      sha224: true,
+    };
+    const security = {
+      policy: 'examplePoolicy',
+      signature: 'blablabla',
+    };
+    return client.metadata(defaultHandle, options, security).then(() => {
+      expect.assertions(2);
+      expect(mockMetadata).toHaveBeenCalledTimes(1);
+      expect(mockMetadata).toHaveBeenCalledWith(client.session, defaultHandle, options, security);
+      done();
+    });
+  });
+  it('should be able to open picker', () => {
+    const client = new Client(defaultApikey);
+    const pickerOptions = {
+      lang: 'de',
+    };
+    client.picker(pickerOptions);
+    expect(mockPicker).toBeCalled();
+    expect(mockPicker).toBeCalledWith(expect.any(Object), pickerOptions);
+  });
+  it('should be able to open preview', () => {
+    const client = new Client(defaultApikey);
+    const previewOptions = {
+      id: 'testElement',
+    };
+    client.preview(defaultHandle, previewOptions);
+    expect(mockPreview).toBeCalled();
+    expect(mockPreview).toBeCalledWith(expect.any(Object), defaultHandle, previewOptions);
+  });
+  it('should be able to remove handle', (done) => {
```

please add another test for remove (should throw an error during remove if security is not given)

---

I only tested logic contained in this file and used mocks for external functions. 
This particular error throw which you mentioned is in ./api/file remove method.